src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Linux.cs

```diff
@@ -78,7 +78,11 @@ public class NetworkChange
                     }
                     if (s_availabilityTimer == null)
                     {
+                        // Don't capture the current ExecutionContext and its AsyncLocals onto the timer causing them to live forever
```

Comment:
Why not?  What's special about this case that the execution context shouldn't flow into the callback?  I can see the argument being made in cases where only infrastructure code is invoked in the callback, but here user code is being invoked.

(And if it is necessary here, the current structure of this is problematic, as it could end up throwing an exception if flow is already suppressed.)

Reply:
Its capturing the asynclocals for only the first added delegate onto the timer; not the second, third etc; so it would be the wrong locals.

However I assume if the EC was going to be captured its the addition of to the delegate that should capture them (with each addition being its own context)
```csharp
s_availabilityChangedSubscribers += value;
```
Not the single creation of a globally static timer?

Reply:
Fixed the suppression